gradio/routes.py

```diff
@@ -202,13 +209,51 @@ def __init__(
         self.auth_dependency = auth_dependency
         self.api_info = None
         self.all_app_info = None
+
         # Allow user to manually set `docs_url` and `redoc_url`
         # when instantiating an App; when they're not set, disable docs and redoc.
         kwargs.setdefault("docs_url", None)
         kwargs.setdefault("redoc_url", None)
         self.custom_component_hashes: dict[str, str] = {}
         super().__init__(**kwargs)
 
+    @staticmethod
+    async def proxy_to_node(
+        request: Request,
+        server_name: str,
+        node_port: int,
+        python_port: int,
+    ) -> Response:
+        client = httpx.AsyncClient()
```

Comment:
I think we can create one client at the top level of the file and re-use it. It may even be better performance as it will use http connection pooling: https://www.python-httpx.org/advanced/clients/

Reply:
Oh wait thats because of the `with` usage, if i change that, it will work.